tests/system/data_sources/test_filesystem.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest import mock
+
+
+from data_validation import (
+    data_validation,
+    consts,
+)
+
+from tests.system.data_sources.common_functions import (
+    schema_validation_test,
+)
+
+
+def mock_get_connection_config(*args):
+    if args[1] in ("parquet-conn", "mock-conn"):
+        return PARQUET_CONN
+    elif args[1] == "csv-conn":
+        return CSV_CONN
+    elif args[1] == "json-conn":
+        return JSON_CONN
+    elif args[1] == "orc-conn":
+        return ORC_CONN
+    else:
+        return PARQUET_CONN
+
+
+CSV_CONN = {
```

Comment:
Nitpicky but constants should really be defined before and functions, i.e. these four constants should move above the `mock_get_connection_config` definition.

Approving anyway, I'll leave it up to you whether to move them or not.